runtime/runtime/src/actions.rs

```diff
@@ -438,6 +442,18 @@ pub(crate) fn action_deploy_contract(
         })?;
     account.code_hash = code.get_hash();
     set_code(state_update, account_id.clone(), &code);
+    if let Some(cache) = cache {
+        // We only precompile contract for Wasmer VM at the moment.
+        if let Some(err) =
+            precompile(code.get_code(), &code.get_hash(), vm_config, cache.deref(), VMKind::Wasmer)
+        {
+            return Err(StorageError::StorageInconsistentState(format!(
```

Comment:
What if the contract doesn't compile? It looks like `precompile` can throw an error.

Reply:
Actually if contract doesn't compile we could do one of the following:
   * disregard error, but it means we'll try to recompile it later on and spend CPU
   * make deployment an error
Second behavior seems to be more correct, as it will also protect our runtime from attempts to recompile later on.
@evgenykuzyakov probably your question was not regarding if we shall return an error, but more liek what kind of error shall be returned here. In review feedback I introduced `DataProcessingError` which tries to distinguish it from the storage error.

Reply:
Regarding protocol change - that's an interesting question. You mean protocol changes because we now fail earlier than before in case of invalid contracts, right? Could you please recommend what else to be done to migrate to the new protocol, but just bumping `PROTOCOL_VERSION` in version.rs?